src/geometry/LinearPropagator.i.hh

```diff
@@ -46,16 +50,20 @@ void LinearPropagator::operator()()
  * \pre Assumes that next_step() has been properly called by client
  */
 CELER_FUNCTION
-void LinearPropagator::operator()(real_type dist)
+LinearPropagator::OutputType LinearPropagator::operator()(real_type dist)
 {
     REQUIRE(dist > 0.);
-    REQUIRE(dist <= track_.next_step());
-    this->apply_linear_step(dist);
+    OutputType result;
+    const real_type minstep = fmin(dist, track_.next_step() + track_.tolerance());
```

Comment:
Yes, I haved tested with CUDA 10.2.  I also tested std::min, but that doesn't work with CUDA.
I also tried vecCore::math::Min(), which works fine, but fmin() seems better as it works.

Reply:
Oh whoops, my test code was calling `std::fmin(value)` instead of `std::fmin(a, b)` 🤦 

This is good to know, thanks! I couldn't find `fmin` among the functions listed in the cuda C manual actually. It does seem that the CUDA function works with the `std` namespace, so could you either add `using std::fmin` inside the function scope, or change `fmin` to `std::fmin`? This makes the code more standards-compliant (robust to changes in compiler implementations).